plugins/fabric8-cdn-support/src/main/java/com/redhat/che/cdnSupport/CdnSupportService.java

```diff
@@ -0,0 +1,315 @@
+
+    LOG.debug("Parse Plugin YAML file: {}", pluginYaml.toAbsolutePath().toString());
+    JsonNode json = YAML_PARSER.readTree(pluginYaml.toFile());
+
+    LOG.debug("Plugin YAML file content: {}", json.toString());
+    return stream(spliteratorUnknownSize(json.path("containers").elements(), 0), false)
+        .findFirst()
+        .map(node -> node.path("image").asText())
+        .orElse(null);
+  }
+
+  static class CommandRunner {
+    Process runCommand(
+        String command,
+        String[] arguments,
+        File directory,
+        long timeout,
+        TimeUnit timeUnit,
+        LineConsumer outputConsumer,
+        LineConsumer errorConsumer)
+        throws IOException, TimeoutException, InterruptedException, ExecutionException {
+      final String[] commandLine = new String[arguments.length + 1];
+      Lists.asList(command, arguments).toArray(commandLine);
+      ProcessBuilder processBuilder = new ProcessBuilder();
+      if (directory != null) {
+        processBuilder.directory(directory);
+      }
+      processBuilder.command(commandLine);
+      LOG.debug("Command: {}", processBuilder.command());
+      LOG.debug("Directory: {}", processBuilder.directory());
+      Process process = processBuilder.start();
+      CompletableFuture<Void> readers =
+          allOf(
+              of(outputConsumer, errorConsumer)
+                  .map(
+                      (LineConsumer consumer) -> {
+                        return runAsync(
+                            () -> {
+                              if (outputConsumer != null) {
+                                try {
+                                  // consume logs until process ends
+                                  process(process, outputConsumer);
+                                } catch (IOException e) {
+                                  LOG.error(
+                                      format(
+                                          "Failed to complete reading of the process '%s' output or error due to occurred error",
+                                          Joiner.on(" ").join(commandLine)),
+                                      e);
+                                }
+                              }
+                            });
+                      })
+                  .collect(toList())
+                  .toArray(FUTURE_ARRAY));
+      try {
+        if (!process.waitFor(timeout, timeUnit)) {
+          try {
+            ProcessUtil.kill(process);
+          } catch (RuntimeException x) {
+            LOG.error(
+                "An error occurred while killing process '{}'", Joiner.on(" ").join(commandLine));
+          }
+          throw new TimeoutException(
+              format(
+                  "Process '%s' was terminated by timeout %s %s.",
+                  Joiner.on(" ").join(commandLine), timeout, timeUnit.name().toLowerCase()));
+        }
+      } finally {
+        readers.get(2, TimeUnit.SECONDS);
```

Comment:
should it be configurable (timeout ? )

Reply:
I don't think it's necessary. It won't ever take more than 2 seconds to flush the standard and error outputs afact